### PR TITLE
feat(security): Cleanup Vault TLS related overrides

### PR DESCRIPTION
**Pull request by jim-wang-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [X ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/developer-scripts/blob/master/.github/Contributing.md.

## What is the current behavior?

Current override values for Vault still using TLS related values but should be cleaned up.

## What is the new behavior?
Vault TLS-related overrides should be cleaned up

## Does this PR introduce a breaking change?

- [ ] Yes
- [X ] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?
**Note:  For this compose up working correctly, it requires the following PR merged into master first:**
1. https://github.com/edgexfoundry/edgex-go/pull/2890 (done)

## Other information
If tested locally, do the following steps:
1. Clone the code from https://github.com/edgexfoundry/edgex-go/pull/2890 
1. Build the latest docker images for `edgex-go`: `make docker`
1. Go to `compose-builder` sub-directory and do `make gen dev` to generate the local version of `docker-compose.yml` to run
1. In the command-line, run `docker-compose up` and all services should started up ok (can be checked and see using `docker ps` and individual docker logs via portainer).   Also it should be ok in the docker logs `edgex-core-consul`

### CODE REVIEW

**Comment by beaufrusetta** [comment]
I'm getting an error @ step 2 when running the testing instructions listed in the PR description.

**Comment by jim-wang-intel** [comment]
> 
> 
> I'm getting an error @ step 2 when running the testing instructions listed in the PR description.
> 
> `Step 21/25 : RUN pip install docker-compose==1.23.2 ---> Running in 4b39d48641e3 /bin/sh: pip: not found The command '/bin/sh -c pip install docker-compose==1.23.2' returned a non-zero code: 127 make: *** [Makefile:161: docker_sys_mgmt_agent] Error 127 `
> 
> Looks like this is when it's attempting to build the sys-mgmt-agent.

Weird, I didn't change anything on that part related to `syst-mgmt-agent` at all.   @beaufrusetta , What's OS you were using to run this?  Can you try to run `make build` to see if it even gets compiled?

**Comment by beaufrusetta** [comment]
Thanks @jim-wang-intel - it works now!

File: compose-builder/add-security.yml [hunk 1]
```diff
@@ -233,8 +232,7 @@ services:
       KONGURL_SERVER: kong
       SECRETSERVICE_SERVER: edgex-vault
       SECRETSERVICE_TOKENPATH: /tmp/edgex/secrets/edgex-security-proxy-setup/secrets-token.json
-      SECRETSERVICE_CACERTPATH: /tmp/edgex/secrets/ca/ca.pem
-      SECRETSERVICE_SNIS: "edgex-kong"
+      SECRETSERVICE_SNIS: ''
```
Comment (lenny-intel) [inline compose-builder/add-security.yml:235]: Can we just remove the override? I.e. edgex-proxy should have this as blank by default if we always have it a blank?

File: compose-builder/common-security.env [hunk 2]
```diff
@@ -21,5 +21,5 @@
 
 EDGEX_SECURITY_SECRET_STORE=true
 SECRETSTORE_HOST=edgex-vault
-SECRETSTORE_SERVERNAME=edgex-vault
-SECRETSTORE_ROOTCACERTPATH=/tmp/edgex/secrets/ca/ca.pem
+SECRETSTORE_SERVERNAME=''
+SECRETSTORE_ROOTCACERTPATH=''
```
Comment (lenny-intel) [inline compose-builder/common-security.env:25]: Now that these are the defaults, they should be blank in the TOML files and removed from here.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: compose-builder/add-security.yml (hunk 1 above)
Comment (jim-wang-intel) [reply compose-builder/add-security.yml:235]: ok

File: compose-builder/common-security.env (hunk 2 above)
Comment (jim-wang-intel) [reply compose-builder/common-security.env]: removed

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
Waiting of edgex-go PR edgexfoundry/edgex-go#2890 before merge.

**Comment by lenny-intel** [comment]
recheck

**Comment by cherrycl** [comment]
recheck

**Review by lenny-intel — APPROVED** [review]

File: releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml [hunk 3]
```diff
@@ -250,13 +246,9 @@ services:
       EDGEX_SECURITY_SECRET_STORE: "true"
       KONGURL_SERVER: kong
       REGISTRY_HOST: edgex-core-consul
-      SECRETSERVICE_CACERTPATH: /tmp/edgex/secrets/ca/ca.pem
       SECRETSERVICE_SERVER: edgex-vault
-      SECRETSERVICE_SNIS: edgex-kong
```
Comment (bnevis-i) [inline releases/nightly-build/compose-files/docker-compose-nexus-arm64.yml:255]: Technically speaking, this is a Kong TLS related override, not a Vault TLS override.

**Pull request merged**